libr/core/cmd_magic.c

```diff
@@ -190,11 +190,14 @@ static int r_core_magic_at(RCore *core, const char *file, ut64 addr, int depth,
 	return ret;
 }
 
-static void r_core_magic(RCore *core, const char *file, int v) {
+static void r_core_magic(RCore *core, const char *file, int v, int json) {
 	ut64 addr = core->offset;
 	int hits = 0;
 	magicdepth = r_config_get_i (core->config, "magic.depth"); // TODO: do not use global var here
-	r_core_magic_at (core, file, addr, magicdepth, v, false, &hits);
+	r_core_magic_at (core, file, addr, magicdepth, v, json, &hits);
+	if (json) {
+		r_cons_newline ();
```

Comment:
You sure you pushed everything?

Reply:
"r_core_magic_at" already supported json output. I just had to add a newline at the end to make sure that the printed json is not cleared

Reply:
ah nice :) Can you add test in json tests?

Reply:
Sure, will add the test